### PR TITLE
Support scala.js on Scala 3

### PR DESCRIPTION
**Pull request by rossabaker**
Follow-up to #63.  Moved the package-platforms to scala-2, because we removed the package object in Scala 3.  In Scala 3, we can freely mix top-level definitions, so don't need the empty JVM stub anymore.

I'll plan to release this as another milestone.

/cc @Kevin-Lee

### CODE REVIEW

**Comment by Kevin-Lee** [comment]
Great! Thank you! 👍

File: core/js/src/main/scala-3/org/log4s/package.scala [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package org.log4s
+
+final val Log4sConfig: log4sjs.Log4sConfig = log4sjs.Log4sConfig
+final val AllThreshold = log4sjs.LogThreshold.AllThreshold
+final val OffThreshold = log4sjs.LogThreshold.OffThreshold
```
Comment (Kevin-Lee) [inline core/js/src/main/scala-3/org/log4s/package.scala:5]: I think if you want to have it here, you need to move the following two files to `core/jvm/src/main/scala-3/org/log4s/package.scala`
* `core/shared/src/main/scala-3.0.0-M1/org/log4s/package.scala`
* `core/shared/src/main/scala-3.0.0-M2/org/log4s/package.scala`

So leave only one and move from `shared` to `jvm`.

File: build.sbt [hunk 2]
```diff
@@ -35,7 +35,7 @@ val prevArtifacts = Def.derive {
 
 def jsOpts = new Def.SettingList(Seq(
   scalaJSLinkerConfig ~= { _.withModuleKind(ModuleKind.CommonJSModule) },
-  crossScalaVersions := crossScalaVersions.value.filter(_.startsWith("2."))
+  crossScalaVersions := crossScalaVersions.value.filterNot(_ == "3.0.0-M1")
```
Comment (Kevin-Lee) [inline build.sbt:38]: Does `3.0.0-M2` have no issues with build for Scala.js?

Comment (Kevin-Lee) [reply build.sbt:38]: I just tired it and after applying what I said [here](https://github.com/Log4s/log4s/pull/64/files#r536001193), it still fails to compile for Scala.js for Scala 3.0.0-M2.

Comment (rossabaker) [reply build.sbt:38]: I thought this compiled last night, but apparently not.  I don't particularly want to block a milestone on sorting this out.  Did you need it for your project, or is the JVM version sufficient?

Comment (Kevin-Lee) [reply build.sbt:38]: Yeah, I need it for my project and the JVM version is sufficient. It would be great, if it can be release. Thank you!

**Comment by rossabaker** [comment]
This is closer, but I get the following on `++3.0.0-RC2 test` (hacking Dependencies.scala is necessary until #73 is merged):

```
[error] Referring to non-existent method scala.scalajs.js.$bar$.UnionOps(java.lang.Object)java.lang.Object
[error]   called from org.log4s.log4sjs.FunctionalType.fromDynamicType(scala.scalajs.js.$bar)scala.scalajs.js.Any
[error]   called from static org.log4s.log4sjs.FunctionalType.org$log4s$log4sjs$FunctionalType$Provider$$$_$$lessinit$greater$$anonfun$4(org.log4s.log4sjs.FunctionalType,scala.scalajs.js.$bar)scala.scalajs.js.Any
[error]   called from constructor org.log4s.log4sjs.FunctionalType$Provider$.<init>(org.log4s.log4sjs.FunctionalType)void
[error]   called from org.log4s.log4sjs.FunctionalType.Provider()org.log4s.log4sjs.FunctionalType$Provider$
[error]   called from org.log4s.PlatformInit$.init()void
[error]   called from org.log4s.LoggerInit.$init$()void
[error]   called from constructor org.log4s.GetLoggerSpec.<init>()void
[error]   called from static constructor org.log4s.GetLoggerSpec.<stinit>()void
[error]   called from core module analyzer
```

I'm not bothering with 3.0.0-RC1, which is old news and doesn't have the scalatestplus-scalacheck library.

**Comment by rossabaker** [comment]
I'd still like to get this one done before Scala 3 is out.

**Comment by rossabaker** [comment]
I'd still like to get this one done before Scala 3 is out.

**Comment by rossabaker** [comment]
Fixed by #75.

**Pull request closed without merging**